deployment/streaming/scripts/batch-process.sh

```diff
@@ -33,18 +34,36 @@ aws emr create-cluster \
 	        \"InstanceCount\": 1,
 	        \"BidPrice\": \"OnDemandPrice\",
 	        \"InstanceGroupType\": \"MASTER\",
-	        \"InstanceType\": \"${BATCH_INSTANCE_TYPE}\",
-	        \"Name\":\"Master\"
+	        \"InstanceType\": \"m4.xlarge\",
+	        \"Name\":\"Master\",
+	        \"EbsConfiguration\": {
+						\"EbsOptimized\": true,
+						\"EbsBlockDeviceConfigs\": [{
+							\"VolumeSpecification\": {
+								\"VolumeType\": \"gp2\",
+								\"SizeInGB\": 1024
+							}
+						}]
+					}
 	      }, {
 	        \"InstanceCount\": ${NUM_EXECUTORS},
 	        \"BidPrice\": \"OnDemandPrice\",
 	        \"InstanceGroupType\": \"CORE\",
-	        \"InstanceType\": \"${BATCH_INSTANCE_TYPE}\",
-	        \"Name\":\"Workers\"
+	        \"InstanceType\": \"r5.xlarge\",
```

Comment:
My train of thought here was that since these instance types are deviating from the default and are specific to this task that they should just be hardcoded as known good instance types for the task. Otherwise we'll eventually just end up with a separate Makefile var for each task, which is effectively the same as just having them hardcoded.

Reply:
That might hold, except that I intended the `batch-process.sh` script to be a general utility for launching any kind of batch job that might come up in the future.  Being able to select the general nature of the cluster and pass the job description quickly was to be its paramount virtue.  I hadn't expected the cluster type to need changing (certainly not so soon), so I think it's reasonable to make it a parameter?

Reply:
As a for instance, if we ever want to set up a make rule that can be used to kick off an `osm2orc` conversion job in a semi-automated fashion, I'd expect to use this script.  (That might require some work to allow for multiple steps to be supplied, though.

Reply:
Oh ok. We had just talked on Slack about just hard coding these instance types, I wasn't aware that this script was intended to be more general. I'll see about getting the best of both worlds here by reverting to environment variables for instance types, but putting the declaration of the env var for this particular task in the make command, rather than sourcing it more generically from the aws-env.mk file.

Reply:
Yes, I was thinking that we just pass it as a parameter from the invocation in the makefile.  This doesn't need to be arbitrarily configurable as much as it needs to be flexible.

Reply:
Updated.

Reply:
I was really thinking that we'd pass the instance type as an argument to the `batch-process.sh`, which would allow us to set different arguments to each batch job.  (We should switch to this batch process script instead of `batch-generate-edit-histograms.sh` and `batch-generate-footprints.sh`.  The thought is that we can then customize the recommended cluster size and instance type on a per-job basis.  For example, if one job requires R-type instances, another requires M-type, with each needing different cluster sizes, then we can essentially recommend the different hardware profiles right in the Makefile.  I can see why we wouldn't necessarily want to do this, though, so I leave it as a judgement call for you.

Reply:
I think you might be looking at the wrong commit or have an out of date view or something?

> I was really thinking that we'd pass the instance type as an argument to the batch-process.sh

That's what I did by passing the instance type values as env vars since we already have env vars for this bit of configuration in other scripts:
- https://github.com/azavea/osmesa/pull/184/commits/1b44ce2a52d14fd18640d9e19d31333869ed7189#diff-ea2c8e95fb7100e915004c792020bd08R37
- https://github.com/azavea/osmesa/pull/184/commits/1b44ce2a52d14fd18640d9e19d31333869ed7189#diff-ea2c8e95fb7100e915004c792020bd08R52
- https://github.com/azavea/osmesa/pull/184/commits/1b44ce2a52d14fd18640d9e19d31333869ed7189#diff-7f5b6dfafcbff22d3391e3049877f9beR106

I don't really want to migrate the other batch scripts to batch-process.sh in this PR, although I think that's likely a good idea based on what I've seen so far. That's additional work that requires testing and moving things around, I made another issue for it: https://github.com/azavea/osmesa/issues/189

Reply:
Aha.  Didn't see https://github.com/azavea/osmesa/commit/1b44ce2a52d14fd18640d9e19d31333869ed7189#diff-7f5b6dfafcbff22d3391e3049877f9beR106 .  That's what I was after.  Might be a bit confusing since it shares a name with what is in the `config-deployment.mk`, but it makes sense why you went that way.